llarp/rpc/rpc_server.cpp

```diff
@@ -107,626 +61,518 @@ namespace llarp::rpc
-                  {
-                    auto src = std::make_shared<DummyPacketSource>([reply](auto result) {
-                      if (result)
-                        reply(CreateJSONResponse(result->ToJSON()));
-                      else
-                        reply(CreateJSONError("no response from dns"));
-                    });
-                    if (not dns->MaybeHandlePacket(src, src->dumb, src->dumb, msg.ToBuffer()))
-                    {
-                      reply(CreateJSONError("dns query not accepted by endpoint"));
-                    }
-                  }
-                  else
-                    reply(CreateJSONError("endpoint does not have dns"));
-                  return;
-                }
-                reply(CreateJSONError("no such endpoint for dns query"));
-              });
-            })
-        .add_request_command("config", [&](oxenmq::Message& msg) {
-          HandleJSONRequest(msg, [r = m_Router](nlohmann::json obj, ReplyFunction_t reply) {
+    if (quiclistener.request.port == 0 and quiclistener.request.closeID == 0)
+    {
+      SetJSONError("Invalid arguments", quiclistener.response);
+      return;
+    }
+
+    auto endpoint = (quiclistener.request.endpoint.empty())
+        ? GetEndpointByName(m_Router, "default")
+        : GetEndpointByName(m_Router, quiclistener.request.endpoint);
+
+    if (not endpoint)
+    {
+      SetJSONError("No such local endpoint found", quiclistener.response);
+      return;
+    }
+
+    auto quic = endpoint->GetQUICTunnel();
+
+    if (not quic)
+    {
+      SetJSONError(
+          "No quic interface available on endpoint " + quiclistener.request.endpoint,
+          quiclistener.response);
+      return;
+    }
+
+    if (quiclistener.request.closeID)
+    {
+      quic->forget(quiclistener.request.closeID);
+      SetJSONResponse("OK", quiclistener.response);
+      return;
+    }
+
+    if (quiclistener.request.port)
+    {
+      auto id = 0;
+      try
+      {
+        SockAddr addr{quiclistener.request.remoteHost, huint16_t{quiclistener.request.port}};
+        id = quic->listen(addr);
+      }
+      catch (std::exception& e)
+      {
+        SetJSONError(e.what(), quiclistener.response);
+        return;
+      }
+
+      util::StatusObject result;
+      result["id"] = id;
+      std::string localAddress;
+      var::visit([&](auto&& addr) { localAddress = addr.ToString(); }, endpoint->LocalAddress());
+      result["addr"] = localAddress + ":" + std::to_string(quiclistener.request.port);
+
+      if (not quiclistener.request.srvProto.empty())
+      {
+        auto srvData = dns::SRVData::fromTuple(
+            std::make_tuple(quiclistener.request.srvProto, 1, 1, quiclistener.request.port, ""));
+        endpoint->PutSRVRecord(std::move(srvData));
+      }
+
+      SetJSONResponse(result, quiclistener.response);
+      return;
+    }
+  }
+
+  void
+  RPCServer::invoke(LookupSnode& lookupsnode)
```

Comment:
this RPC call should be revisited (i.e. what does/should it do and why?) but that may be out of scope for this PR

Reply:
agreed. only included it as it was present pre-refactor